sign_up/sign_up.js

```diff
@@ -22,43 +19,90 @@ formElem.addEventListener("submit", (event) => {
   event.preventDefault();
 });
 
-const usersState = {
+const userData = {
   name: null,
   surname: null,
   email: null,
   password: null,
-  confirmPwd: null,
 };
 
 const inputs = document.querySelectorAll("input");
 const confirmPassword = document.getElementById("confirm-password");
 const signUpBtn = document.getElementById("sign-up-btn");
 
 signUpBtn.addEventListener("click", () => {
-  if (usersState.password !== usersState.confirmPwd) {
+  if (userData.password !== confirmPassword.value) {
     dialogWarning.innerHTML = "Please recheck your password!";
     dialog.classList.add("active");
     overlay.classList.add("active");
   }
+  api.signedUp(userData);
+  body.innerHTML = "";
+  const welcomeBox = document.createElement("div");
+  welcomeBox.classList.add("welcome-box");
+  const welcomeText = document.createElement("h1");
+  welcomeText.innerHTML = `Welcome, ${userData.name}!`;
+  const signOutBtn = document.createElement("button");
+  signOutBtn.innerHTML = "Sign Out";
+  signOutBtn.classList.add("sign-out-btn");
+  welcomeBox.append(welcomeText);
+  welcomeBox.append(signOutBtn);
+  body.append(welcomeBox);
+  signOutBtn.addEventListener("click", () => {
+    api.signedOut(userData);
+  });
 });
 
-function makeInputValueHandler(usersState) {
-  function takeAndUpdateInputValue(event) {
+function makeInputChangeHandler(category) {
+  function inputChangeHandler(event) {
     const filledUpName = event.target.value;
-    for (let category in usersState) {
+    if (event.target.name !== "confirmPwd") {
       if (category === event.target.name) {
-        usersState[category] = filledUpName;
+        userData[category] = filledUpName;
       }
     }
   }
 
-  return takeAndUpdateInputValue;
+  return inputChangeHandler;
 }
 
-const takeAndUpdateInputValue = makeInputValueHandler(usersState);
-
-function registerNewUser() {
+function addInputHandlers() {
   inputs.forEach((input) => {
-    input.addEventListener("change", takeAndUpdateInputValue);
+    const inputChangeHandler = makeInputChangeHandler(input.name);
+    input.addEventListener("change", inputChangeHandler);
   });
 }
+
+function backend() {
+  const usersDataBase = [];
+  return {
+    getUser(userData) {
+      // can't sign up two times to get two users' info
+    },
+    signedUp(userData) {
+      usersDataBase.push(userData);
+      userData.isSignedUp = true;
+      userData.id = usersDataBase[userData];
+      userData.id = usersDataBase.indexOf(userData);
+      delete userData.password;
+    },
+    signedIn(userData) {
+      // checking email and password
+      userData.isSignedIn = true;
+      window.location.href = "../homepage/index.html";
+    },
+    signedOut(userData) {
+      userData.isSignedIn = false;
```

Comment:
you probably want to reset the entire userData on the front end, not just the isSignedIn flag.